### PR TITLE
made remote_path default and updated docstring

### PR DESCRIPTION
**Pull request by krishnaduttPanchagnula**
This solves the issue number #281

### CODE REVIEW

**Comment by kbolashev** [comment]
Hi Krishna!

Thank you very much for the PR!
I understand that you've implemented the PR according to what's been said in the issue, however we try to have unsurprising defaults where it's possible. This way the users don't have to provide unnecessary parameters in "happy paths" that can be guessed.
I think with this function the better way to go would be to, instead of forcing the users to input the `remote_path`parameter, it'd be better to check if it's None and then set it for the user.
This is actually already done for file uploading here https://github.com/DagsHub/client/blob/7138e44350f778a1d48ec77f36ead475d692ade6/dagshub/upload/wrapper.py#L516
But for the case of folder uploads it's not being done and it breaks.

**Comment by krishnaduttPanchagnula** [comment]
@kbolashev I have added a check to see if the file path is none. Nad if it is none then we are assigning our default path 

DEFAULT_REMOTE_PATH = "/data/src/".

Please comment changes if any.

File: dagshub/upload/wrapper.py [hunk 1]
```diff
@@ -194,10 +195,10 @@ def upload(
 
         """
         if os.path.isdir(local_path):
-            dir_to_upload = self.directory(remote_path)
+            dir_to_upload = self.directory(remote_path if remote_path is not None else DEFAULT_REMOTE_PATH)
             dir_to_upload.add_dir(local_path, commit_message=commit_message, **kwargs)
         else:
-            file_to_upload = DataSet.get_file(local_path, remote_path)
+            file_to_upload = DataSet.get_file(local_path, remote_path if remote_path is not None else DEFAULT_REMOTE_PATH)
             self.upload_files([file_to_upload], commit_message=commit_message, **kwargs)
```
Comment (kbolashev) [inline dagshub/upload/wrapper.py:202]: The `get_file` function already handles the `remote_path is None` case. 
It gets the filename from the `local_path` and uploads with that. So this is not required + it's going to always write the file at `/data/src/`

File: dagshub/upload/wrapper.py [hunk 2]
```diff
@@ -194,10 +195,10 @@ def upload(
 
         """
         if os.path.isdir(local_path):
-            dir_to_upload = self.directory(remote_path)
+            dir_to_upload = self.directory(remote_path if remote_path is not None else DEFAULT_REMOTE_PATH)
             dir_to_upload.add_dir(local_path, commit_message=commit_message, **kwargs)
```
Comment (kbolashev) [inline dagshub/upload/wrapper.py:199]: This is perfectly fine and thank you for actually doing this :)

I think for this to work perfectly it'd be cool to get the directory name, and upload to `/data/{name_of_the_directory}`.

BTW, I'm going to work a little bit on the Repo class in general, so I might implement the behavior that the upload happens relative to the repo root (if the script is called in a git repository).
So, for example, if you have your repo in `~/repos/cool-repo/`, do a script with cwd set somewhere inside of the repo AND then you upload a folder `~/repos/cool-repo/some/other/folder/with/data`, we'll upload the folder to `some/other/folder/with/data`.
BUT, if cwd is not inside of a repository, then upload it to the `/data/src/` folder just like you implemented. Let me know what you think about it, I'd love to hear your thoughts :)

**Review by kbolashev — CHANGES_REQUESTED** [review]
Leave the `is not None` check only on the directory, and then I'll merge :)
Thank you for coming back to this!!

**Comment by kbolashev** [comment]
Thank you very much for the PR! 🥳

File: dagshub/upload/wrapper.py (hunk 2 above)
Comment (krishnaduttPanchagnula) [reply dagshub/upload/wrapper.py:199]: Actually thats a cool idea! @kbolashev

**Pull request merged**